### PR TITLE
Pardo/create third person camera

### PR DESCRIPTION
**Pull request by mfpardotrain**
Fixes .gitignore
Adds third person camera
Adds Fridge object

### CODE REVIEW

**Review by owencummings — APPROVED** [review]
Send it

File: .godot/editor/project_metadata.cfg
[the remark this answers is not in the document]
Comment (collindaloisio) [reply .godot/editor/project_metadata.cfg:1]: No

File: Prefabs/fridge.mtl
Comment (collindaloisio) [inline Prefabs/fridge.mtl:1]: Nice blend, bro!

File: Scripts/MultiplayerController.cs [hunk 1]
```diff
@@ -20,22 +18,21 @@ public override void _Process(double delta)
 	{
 	}
 	
-	private void PeerConnected(long id){
+	private void PeerConnected(long id) {
 		GD.Print("Player Connected! " + id.ToString());
```
Comment (collindaloisio) [inline Scripts/MultiplayerController.cs:22]: Long term we should standardize an internal logging API. This is fine for now.

File: Scripts/MultiplayerController.cs [hunk 2]
```diff
@@ -20,22 +18,21 @@ public override void _Process(double delta)
...
 	}
 	
-	private void PeerDisconnected(long id){}
+	private void PeerDisconnected(long id) {}
 	
-	private void ConnectedToServer(){
+	private void ConnectedToServer() {
 		long id = Multiplayer.GetUniqueId();
```
Comment (collindaloisio) [inline Scripts/MultiplayerController.cs:28]: I think there are multiple spots in this Class that reference GetUniqueId(). Does it make sense to do it on every physics render?

File: Scripts/Player.cs [hunk 3]
```diff
@@ -40,52 +41,47 @@ public override void _PhysicsProcess(double delta)
...
-			{
+	private float _sensX = 0.25f;
+
+	public override void _Input(InputEvent @event) {
+		if(GetNode<MultiplayerSynchronizer>("MultiplayerSynchronizer").GetMultiplayerAuthority() == Multiplayer.GetUniqueId()) {
+			if (@event.IsActionPressed("jump") && IsOnFloor()) {
 				toJump = true;
 			}
+			if (@event is InputEventMouseMotion eventMouseMotion) {
+				RotateY(Mathf.DegToRad(-eventMouseMotion.Relative.X * _sensX));
```
Comment (collindaloisio) [inline Scripts/Player.cs:74]: what does this do? 

var degrees_looked = RotateY(Mathf.DegToRad(-eventMouseMotion.Relative.X * _sensX));

might make this more readable

File: Scripts/Player.cs [hunk 4]
```diff
@@ -40,52 +41,47 @@ public override void _PhysicsProcess(double delta)
...
+			}
+			if (@event.IsActionPressed("escape")) {
+					GetTree().Quit();
+			}
+			if (@event.IsActionPressed("rotate_right")) {
+				RotateObjectLocal(new Vector3(0, -1, 0), .1f);
```
Comment (collindaloisio) [inline Scripts/Player.cs:80]: vector.UP?

File: Scripts/PlayerCamera.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	}
+
+	// Called every frame. 'delta' is the elapsed time since the previous frame.
+	public override void _Process(double delta) {
+		GlobalPosition = Focus.Position;
+	}
+
+	public override void _Input(InputEvent @event) {
+		if (@event is InputEventMouseMotion eventMouseMotion) {
+				RotateX(Mathf.DegToRad(-eventMouseMotion.Relative.Y * .25f));
```
Comment (collindaloisio) [inline Scripts/PlayerCamera.cs:20]: Seen this twice

File: Scripts/PlayerInfo.cs [hunk 6]
```diff
@@ -1,8 +1,13 @@
-public record PlayerInfo
+using System.Data.Common;
+using Godot;
+
+public record PlayerInfo 
 {
-    public int Id;
+    public int Id { get; set; }
```
Comment (collindaloisio) [inline Scripts/PlayerInfo.cs:6]: I love properties

File: Scripts/PlayerInfo.cs [hunk 7]
```diff
@@ -1,8 +1,13 @@
...
 
-    public PlayerInfo(int i){
-        Id = i;
+    public PlayerInfo() { }
+    public PlayerInfo(int id, Vector3 position) {
```
Comment (collindaloisio) [inline Scripts/PlayerInfo.cs:9]: uhhhhhhhh. See what position my position is in.

**Review by collindaloisio — APPROVED** [review]

**Review by collindaloisio — APPROVED** [review]

**Pull request merged**